src/state/users/reducers.js

```diff
@@ -1,15 +1,57 @@
 import { fromJS, Map } from 'immutable';
 import { combineReducers } from 'redux-immutable';
 
-import { USER_RECEIVE } from 'console/state/action-types';
+import {
+  RECIPE_HISTORY_RECEIVE,
+  RECIPE_PAGE_RECEIVE,
+  USER_RECEIVE,
+} from 'console/state/action-types';
 
 function items(state = new Map(), action) {
   switch (action.type) {
-    case USER_RECEIVE:
+    case USER_RECEIVE: {
       return state.set(action.user.id, fromJS(action.user));
+    }
 
-    default:
+    case RECIPE_HISTORY_RECEIVE: {
+      const revisions = fromJS(action.revisions);
+
+      return state.withMutations(mutableState => {
+        revisions.forEach(revision => {
+          const creator = revision.getIn(['approval_request', 'creator']);
+          if (creator) {
+            mutableState.set(creator.get('id'), creator);
+          }
+
+          const approver = revision.getIn(['approval_request', 'approver']);
+          if (approver) {
+            mutableState.set(approver.get('id'), approver);
```

Comment:
It took me the longest time to understand this code. I even ran it with a bunch of convoluted console.log statement. Eventually, it clicked that the file I'm looking at is state/USERS/reducers.js and not that of recipes. Sigh. Quarter-to-end-of-day.